tempesta_fw/t/functional/tempesta_test.py

```diff
@@ -0,0 +1,89 @@
+import unittest
+from helpers import shell, remote
+
+TEST_DISABLED = "disabled"
+TEST_PENDING = "pending"
+TEST_SUCCESS = "success"
+TEST_FAIL = "fail"
+TEST_ERROR = "error"
+
+class TempestaTestResult(unittest.TextTestResult):
+    matcher = None
+
+    def addError(self, test, err):
+        unittest.TextTestResult.addError(self, test, err)
+
+    def addExpectedFailure(self, test, err):
+        unittest.TextTestResult.addExpectedFailure(self, test, err)
+
+    def addFailure(self, test, err):
+        unittest.TextTestResult.addFailure(self, test, err)
+
+    def addSkip(self, test, reason):
+        unittest.TextTestResult.addSkip(self, test, reason)
+
+    def addSuccess(self, test):
+        unittest.TextTestResult.addSuccess(self, test)
+
+    def addUnexpectedSuccess(self, test):
+        unittest.TextTestResult.addUnexpectedSuccess(self, test)
+
+class TempestaTest(object):
+    test = None
+    state = TEST_DISABLED
+    message = ""
+    name = ""
+
+    def __init__(self, test):
+        self.test = test
+        self.name = test.id()
+        self.reset()
+
+    def reset(self):
+        self.state = TEST_PENDING
+        self.message = ""
+
+    def disable(self):
+        self.state = TEST_DISABLED
+        self.message = "Test is disabled"
+
+class TempestaTestEngine(object):
+
+    def __init__(self, v_level=1):
+        all_tests = []
+        loader = unittest.TestLoader()
+        shell.testsuite_flatten(all_tests, loader.discover('.'))
+        self.v_level = v_level
+        self.tests = [TempestaTest(t) for t in all_tests]
+        remote.connect()
```

Comment:
How about nofile limit? Should it be defined here?  https://github.com/tempesta-tech/tempesta/blob/43e35e9742b7f82adb1411d94a5bf7ff4d310c17/tempesta_fw/t/functional/run_tests.py#L153